src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java

```diff
@@ -165,20 +167,51 @@ public void addRoles_InternalServerError() throws Exception {
 
+        Response responseMock = mock(Response.class);
+        when(responseMock.status()).thenReturn(500);
+
+        when(userProfileRepositoryMock.findByIdamId(any(String.class))).thenReturn(Optional.ofNullable(userProfile));
+        when(idamFeignClientMock.addUserRoles(updateUserProfileData.getRolesAdd(), "1234")).thenReturn(responseMock);
+
+        UserProfileRolesResponse response = sut.updateRoles(updateUserProfileData, userProfile.getIdamId());
+        assertThat(response.getRoleAdditionResponse().getIdamStatusCode()).isEqualTo("500");
+
+        verify(userProfileRepositoryMock, times(1)).findByIdamId(any(String.class));
+        verify(idamFeignClientMock, times(1)).addUserRoles(any(), any(String.class));
+        verify(responseMock, times(1)).close();
+    }
+
+    @Test
+    public void addRoles_FeignException() {
+        RoleName roleName1 = new RoleName("pui-case-manager");
+        RoleName roleName2 = new RoleName("pui-case-organisation");
+        Set<RoleName> roles = new HashSet<>();
+        roles.add(roleName1);
+        roles.add(roleName2);
+
+        updateUserProfileData.setRolesAdd(roles);
+
+        UserProfileResponse userProfileResponse = new UserProfileResponse();
+        RoleAdditionResponse roleAdditionResponse = new RoleAdditionResponse();
+        roleAdditionResponse.setIdamStatusCode(HttpStatus.INTERNAL_SERVER_ERROR.toString());
+        roleAdditionResponse.setIdamMessage("Failure");
+
+        FeignException feignExceptionMock = Mockito.mock(FeignException.class);
+        when(feignExceptionMock.status()).thenReturn(IdamServiceImplTest.StatusCode.INTERNAL_SERVER_ERROR.getStatus());
         when(userProfileRepositoryMock.findByIdamId(any(String.class))).thenReturn(Optional.ofNullable(userProfile));
-        when(idamFeignClientMock.addUserRoles(updateUserProfileData.getRolesAdd(), "1234")).thenReturn(Response.builder().request(mock(Request.class)).body(body, Charset.defaultCharset()).status(500).build());
+        when(idamFeignClientMock.addUserRoles(updateUserProfileData.getRolesAdd(), "1234")).thenThrow(feignExceptionMock);
 
         UserProfileRolesResponse response = sut.updateRoles(updateUserProfileData, userProfile.getIdamId());
         assertThat(response.getRoleAdditionResponse().getIdamStatusCode()).isEqualTo("500");
 
         verify(userProfileRepositoryMock, times(1)).findByIdamId(any(String.class));
-        verify(idamFeignClientMock, times(1)).addUserRoles(any(),any(String.class));
+        verify(idamFeignClientMock, times(1)).addUserRoles(any(), any(String.class));
+        verify(auditServiceMock, times(1)).persistAudit(any(HttpStatus.class), any(UserProfile.class), any(ResponseSource.class));
     }
 
     @Test
     public void deleteRoles_InternalServerError() throws Exception {
         RoleName roleName1 = new RoleName("pui-case-manager");
-        // RoleName roleName2 = new RoleName("pui-case-organisation");
         Set<RoleName> roles = new HashSet<>();
```

Comment:
prefix test for method

Reply:
remove commented code

Reply:
This has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/385/commits/04d04c3144cda8c41c2c8b875f3a62b0fa855501